test/unit/tools/unifed-utils.test.ts

```diff
@@ -0,0 +1,339 @@
+      const parsedProviders = mergeKMSProviders({}, {});
+      expect(parsedProviders).not.to.have.property('aws');
+    });
+
+    it('configures the provider without credentials if an empty object is supplied', function () {
+      const parsedProviders = mergeKMSProviders(
+        {
+          aws: {}
+        },
+        {}
+      );
+      expect(parsedProviders.aws).deep.equal({});
+    });
+
+    it('replaces a $$placeholder value with the value from the environment', function () {
+      const parsedProviders = mergeKMSProviders(
+        {
+          aws: {
+            accessKeyId: { $$placeholder: 1 },
+            secretAccessKey: 'secretAccessKey'
+          }
+        },
+        {
+          aws: { accessKeyId: 'accessKeyId' }
+        }
+      );
+      expect(parsedProviders.aws).deep.equal({
+        accessKeyId: 'accessKeyId',
+        secretAccessKey: 'secretAccessKey'
+      });
+    });
+
+    it('omits required fields if the field is not present in the kmsProviders', function () {
+      const parsedProviders = mergeKMSProviders(
+        {
+          aws: {
+            accessKeyId: { $$placeholder: 1 }
+          }
+        },
+        {
+          aws: { accessKeyId: 'accessKeyId' }
+        }
+      );
+      expect(parsedProviders.aws).not.to.have.property('secretAccessKey');
+    });
+
+    it('configures the provider with the exact credentials from the test', function () {
+      const parsedProviders = mergeKMSProviders(
+        {
+          aws: {
+            accessKeyId: 'accessKeyId',
+            secretAccessKey: 'secretAccessKey',
+            sessionToken: 'sessionToken'
+          }
+        },
+        {
+          aws: {
+            accessKeyId: 'accessKeyIdFromEnvironment',
+            secretAccessKey: 'secretAccessKeyFromEnvironment',
+            sessionToken: 'sessionTokenFromEnvironment'
+          }
+        }
+      );
+      expect(parsedProviders.aws).deep.equal({
+        accessKeyId: 'accessKeyId',
+        secretAccessKey: 'secretAccessKey',
+        sessionToken: 'sessionToken'
+      });
+    });
+  });
+  context('local providers', function () {
+    it('does not configure the provider if none is given', function () {
+      const parsedProviders = mergeKMSProviders({}, {});
+      expect(parsedProviders).not.to.have.property('local');
+    });
+
+    it('configures the provider without credentials if an empty object is supplied', function () {
+      const parsedProviders = mergeKMSProviders(
+        {
+          local: {}
+        },
+        {}
+      );
+      expect(parsedProviders.local).deep.equal({});
+    });
+
+    it('replaces a $$placeholder value with the value from the environment', function () {
+      const parsedProviders = mergeKMSProviders(
+        {
+          local: {
+            key: { $$placeholder: 1 }
+          }
+        },
+        {
+          local: { key: 'key' }
+        }
+      );
+      expect(parsedProviders.local).deep.equal({
```

Comment:
just an fyi in case it's helpful, since you use `property` elsewhere, you *can* use the property assertion with the deep flag set for deep equality comparison: `expect(parsedProviders).to.have.deep.property('local', { key: 'key' })`

but what you're doing here is perfectly fine since the test cases are already under a "local providers" heading so there shouldn't be any confusion about which property match is failing if the test fails

Reply:
I'll leave this as-is, unless you want me to change it

Reply:
nope, this was just an fyi